src/Agent.Listener/LocalRunner.cs

```diff
@@ -21,45 +21,60 @@ namespace Microsoft.VisualStudio.Services.Agent.Listener
     [ServiceLocator(Default = typeof(LocalRunner))]
     public interface ILocalRunner : IAgentService
     {
-        Task<int> RunAsync(CommandSettings command, CancellationToken token);
+        Task<int> LocalRunAsync(CommandSettings command, CancellationToken token);
     }
 
     public sealed class LocalRunner : AgentService, ILocalRunner
     {
-        private readonly Dictionary<string, TaskDefinition> _queryCache = new Dictionary<string, TaskDefinition>(StringComparer.OrdinalIgnoreCase);
-        private Dictionary<string, List<TaskDefinition>> _availableTasks;
         private string _gitPath;
-        private TaskAgentHttpClient _httpClient;
+        private ITaskStore _taskStore;
         private ITerminal _term;
 
         public sealed override void Initialize(IHostContext hostContext)
         {
             base.Initialize(hostContext);
+            _taskStore = HostContext.GetService<ITaskStore>();
             _term = hostContext.GetService<ITerminal>();
         }
 
-        public async Task<int> RunAsync(CommandSettings command, CancellationToken token)
+        public async Task<int> LocalRunAsync(CommandSettings command, CancellationToken token)
         {
-            Trace.Info(nameof(RunAsync));
-            var configStore = HostContext.GetService<IConfigurationStore>();
-            AgentSettings settings = configStore.GetSettings();
+            Trace.Info(nameof(LocalRunAsync));
 
-            // Store the HTTP client.
-            // todo: fix in master to allow URL to be empty and then rebase on master.
-            const string DefaultUrl = "http://127.0.0.1/local-runner-default-url";
-            string url = command.GetUrl(DefaultUrl);
-            if (!string.Equals(url, DefaultUrl, StringComparison.Ordinal))
+            // Warn preview.
+            _term.WriteLine("This command is currently in preview. The interface and behavior will change in a future version.");
+            if (!command.Unattended)
             {
-                var credentialManager = HostContext.GetService<ICredentialManager>();
-                string authType = command.GetAuth(defaultValue: Constants.Configuration.Integrated);
-                ICredentialProvider provider = credentialManager.GetCredentialProvider(authType);
-                provider.EnsureCredential(HostContext, command, url);
-                _httpClient = new TaskAgentHttpClient(new Uri(url), provider.GetVssCredentials(HostContext));
+                _term.WriteLine("Press Enter to continue.");
+                _term.ReadLine();
+            }
+
+            HostContext.RunMode = RunMode.Local;
+
+            // Resolve the YAML file path.
+            string ymlFile = command.GetYml();
+            if (string.IsNullOrEmpty(ymlFile))
+            {
+                string[] ymlFiles =
+                    Directory.GetFiles(Directory.GetCurrentDirectory())
```

Comment:
this seemed like a good idea to fallback to searching for exactly one in the current dir. msbuild does this.